sources/directconnect/virturl-interface.go

```diff
@@ -0,0 +1,187 @@
+package directconnect
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/directconnect"
+	"github.com/overmindtech/aws-source/sources"
+	"github.com/overmindtech/sdp-go"
+)
+
+var gatewayIDVirtualInterfaceIDFmt = "gateway_id:%s virtual_interface_id:%s"
+
+func virtualInterfaceOutputMapper(_ context.Context, _ *directconnect.Client, scope string, _ *directconnect.DescribeVirtualInterfacesInput, output *directconnect.DescribeVirtualInterfacesOutput) ([]*sdp.Item, error) {
+	items := make([]*sdp.Item, 0)
+
+	for _, virtualInterface := range output.VirtualInterfaces {
+		attributes, err := sources.ToAttributesCase(virtualInterface, "tags")
+		if err != nil {
+			return nil, err
+		}
+
+		item := sdp.Item{
+			Type:            "directconnect-virtual-interface",
+			UniqueAttribute: "virtualInterfaceId",
+			Attributes:      attributes,
+			Scope:           scope,
+			Tags:            tagsToMap(virtualInterface.Tags),
+		}
+
+		if virtualInterface.ConnectionId != nil {
+			// +overmind:link directconnect-connection
+			item.LinkedItemQueries = append(item.LinkedItemQueries, &sdp.LinkedItemQuery{
+				Query: &sdp.Query{
+					Type:   "directconnect-connection",
+					Method: sdp.QueryMethod_GET,
+					Query:  *virtualInterface.ConnectionId,
+					Scope:  scope,
+				},
+				BlastPropagation: &sdp.BlastPropagation{
+					// We cannot delete a connection if it has virtual interfaces
+					In: true,
+					// We can't affect the connection
+					Out: false,
+				},
+			})
+		}
+
+		if virtualInterface.DirectConnectGatewayId != nil {
+			// +overmind:link directconnect-direct-connect-gateway
+			item.LinkedItemQueries = append(item.LinkedItemQueries, &sdp.LinkedItemQuery{
+				Query: &sdp.Query{
+					Type:   "directconnect-direct-connect-gateway",
+					Method: sdp.QueryMethod_GET,
+					Query:  *virtualInterface.DirectConnectGatewayId,
+					Scope:  "global",
+				},
+				BlastPropagation: &sdp.BlastPropagation{
+					// We cannot delete a direct connect gateway if it has virtual interfaces
+					In: true,
+					// We can't affect the direct connect gateway
+					Out: false,
+				},
+			})
+		}
+
+		if virtualInterface.AmazonAddress != nil {
+			// +overmind:link ip
+			item.LinkedItemQueries = append(item.LinkedItemQueries, &sdp.LinkedItemQuery{
+				Query: &sdp.Query{
+					Type:   "ip",
```

Comment:
The IP source [doesn't support a GET using a CIDR](https://github.com/overmindtech/stdlib-source/blob/main/sources/network/ip.go#L13), since it returns details of a single IP address. However the [`rdap-ip-network` does](https://github.com/overmindtech/stdlib-source/blob/main/sources/internet/ip_network.go#L14C19-L15C1), and gets details of that network block. Can we change this to do a `SEARCH` against that type instead?

Reply:
https://github.com/overmindtech/aws-source/pull/373/commits/c8a30216886144b7f95ea9d13d8b8d4f49e808be